crates/shared/src/token_list.rs

```diff
@@ -62,6 +92,14 @@ struct TokenModel {
     token: Token,
 }
 
+#[async_trait::async_trait]
+impl Maintaining for TokenList {
+    async fn run_maintenance(&self) -> Result<()> {
+        *self.tokens.lock().unwrap() = self.configuration.tokens().await?;
```

Comment:
Does this hold the lock while awaiting tokens to be fetched or is the optimiser smart enough to only acquire the lock after?

Reply:
Does not hold the lock. Confirmed with the help of a unit test I did not commit since it's not meaningful.